src/components/NavigationFrame.js

```diff
@@ -67,22 +67,10 @@ export default function NavigationFrame({ children }) {
 
 function NetworkSelector() {
   const { endpoint, setEndpoint } = useConnectionConfig();
+  const cluster = useMemo(() => clusterForEndpoint(endpoint), [endpoint])
```

Comment:
I think we could do better but that would require changing a bit the connection context, not sure if it is worth it.